oauth2-server/src/main/java/com/clouway/oauth2/IssueNewTokenActivity.java

```diff
@@ -43,29 +43,31 @@
 
   @Override
   public Response execute(Client client, Identity identity, Set<String> scopes, Request request, DateTime instant) {
-    Map<String, Block> certificates = keyStore.privateCertificates();
-    String certKey = randomKey(certificates);
-    Pem.Block key = certificates.get(certKey);
-
     TokenResponse response = tokens.issueToken(GrantType.AUTHORIZATION_CODE, client, identity, scopes, instant);
     if (!response.isSuccessful()) {
       return OAuthError.invalidRequest("Token cannot be issued.");
     }
 
     BearerToken accessToken = response.accessToken;
-    String host = request.header("Host");
-
-    String idToken = Jwts.builder()
-            .setHeaderParam("cid", certKey)//CertificateId - the ID of the certificate that the token was signed with.
-            .setIssuer(host)
-            .setAudience(client.id)
-            .setSubject(identity.id())
-            .setIssuedAt(instant.asDate())
-            .setExpiration(new Date(accessToken.expirationTimestamp()))
-            .claim("identity", identityAsJson(identity).toString())
-            .signWith(SignatureAlgorithm.RS256, parsePem(key))
-            .compact();
-
+    Map<String, Block> certificates = keyStore.privateCertificates();
+    String idToken = "";
+
+    if (available(certificates)) {
```

Comment:
I'm not sure that null check need to be added. It's bad practices methods to return null but we cannot relay on correct behaviour from client apps. 

if is readable if it's inline:

```java
if (certificates != null && !certificates.isEmpty()) {

}
```